src/test/java/org/broadinstitute/hellbender/engine/dataflow/transforms/examples/SampleUnitTest.java

```diff
@@ -0,0 +1,132 @@
+        s.add(KV.of(new ReferenceShard(2, "2"), 2));
+        s.add(KV.of(new ReferenceShard(3, "3"), 3));
+        s.add(KV.of(new ReferenceShard(4, "4"), 4));
+
+        return new Object[][]{
+                { integers, s}
+        };
+    }
+
+    @DataProvider(name = "refs")
+    public Object[][] refs(){
+        ReferenceShard[] integers = new ReferenceShard[]{
+                new ReferenceShard(1, "1"),
+                new ReferenceShard(2, "2"),
+                new ReferenceShard(3, "3"),
+                new ReferenceShard(4, "4")
+        };
+        Set<ReferenceShard> s = new HashSet<>();
+        s.add(new ReferenceShard(1, "1"));
+        s.add(new ReferenceShard(2, "2"));
+        s.add(new ReferenceShard(3, "3"));
+        s.add(new ReferenceShard(4, "4"));
+
+        return new Object[][]{
+                { integers, s}
+        };
+    }
+
+    static class GroupIntsDoFn extends DoFn<Integer, KV<ReferenceShard, Integer>> {
+        private static final long serialVersionUID = 1L;
+        @Override
+        public void processElement(ProcessContext c) throws Exception {
+            Integer i = c.element();
+            c.output(KV.of(new ReferenceShard(i.intValue(), i.toString()), i));
+        }
+    }
+
+    static class IterableRefShardsDoFn extends DoFn<ReferenceShard, Iterable<ReferenceShard>> {
+        private static final long serialVersionUID = 1L;
+        @Override
+        public void processElement(ProcessContext c) throws Exception {
+            ReferenceShard i = c.element();
+            c.output(Lists.newArrayList(i, i, i));
+        }
+    }
+
+    @Test(dataProvider = "refAndInts")
+    public void GroupIntsTest(Integer[] integers, Set<KV<ReferenceShard, Integer>> s) {
+        DoFnTester<Integer, KV<ReferenceShard, Integer>> fnTester = DoFnTester.of(new GroupIntsDoFn());
+        List<KV<ReferenceShard, Integer>> kvs = fnTester.processBatch(integers);
+        for (KV<ReferenceShard, Integer> r : kvs) {
+            Assert.assertTrue(s.contains(r));
+        }
+    }
+
+    @Test(dataProvider = "refs")
+    public void anotherDoFnTest(ReferenceShard[] referenceShards, Set<ReferenceShard> s) {
+        DoFnTester<ReferenceShard, Iterable<ReferenceShard>> fnTester = DoFnTester.of(new IterableRefShardsDoFn());
+        List<Iterable<ReferenceShard>> kvs = fnTester.processBatch(referenceShards);
+
+        Assert.assertEquals(referenceShards.length, kvs.size());
+        for (Iterable<ReferenceShard> intz : kvs) {
+            int size = 0;
+            for (ReferenceShard i : intz) {
+                size++;
+                Assert.assertTrue(s.contains(i));
+            }
+            Assert.assertEquals(3, size);
+        }
+    }
+
+    @Test(dataProvider = "refs")
+    public void createRefsTest(ReferenceShard[] referenceShards, Set<ReferenceShard> s) {
+        // The simplist way to figure out if a class is coded correctly is to create a PCollection
```

Comment:
I think you meant "simplest". I'm also not entirely sure that this really works. Did you try it with a buggy coder?